androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/SeekClickableArea.kt

```diff
@@ -0,0 +1,81 @@
+package com.profusion.androidenhancedvideoplayer.components.playerOverlay
+
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.indication
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import com.profusion.androidenhancedvideoplayer.R
+import com.profusion.androidenhancedvideoplayer.styling.Dimensions
+
+private const val TEXT_MAX_LINES = 2
+
+@Composable
+fun SeekClickableArea(
+    modifier: Modifier = Modifier,
+    scaleAnimation: Float,
+    tapCount: Int,
+    disableSeekClick: Boolean = false,
+    onSeekSingleTap: () -> Unit,
+    onSeekDoubleTap: () -> Unit,
+    checkIfCanToggleIsControlsVisible: () -> Unit,
+    getSeekTime: () -> Int,
+    seekIcon: @Composable (modifier: Modifier) -> Unit
+) {
+    val isTapCountGreaterThanZero = tapCount > 0
+    Box(
+        modifier = Modifier
+            .fillMaxHeight()
+            .indication(
+                indication = null,
+                interactionSource = remember { MutableInteractionSource() }
+            )
+            .pointerInput(tapCount, disableSeekClick) {
+                detectTapGestures(
+                    onTap = if (isTapCountGreaterThanZero) {
+                        if (disableSeekClick) {
+                            null
+                        } else { { onSeekSingleTap() } }
+                    } else {
+                        { checkIfCanToggleIsControlsVisible() }
+                    },
+                    onDoubleTap = if (isTapCountGreaterThanZero) { null } else {
+                        if (disableSeekClick) {
+                            null
+                        } else {
+                            { onSeekDoubleTap() }
+                        }
+                    }
+                )
```

Comment:
nit: now you have more than one state, so it would be better to use when instead of if 😅

Reply:
now i need the if because this additional state needs to be nested 🤣

Reply:
note that the check need to happen only when the `isTapCountGreaterThanZero` is true